### PR TITLE
oc_erchef users list: allow filtering by external_authentication_id

### PR DESCRIPTION
**Pull request by sdelano**
For the enablement of external authentication mechanisms that are not LDAP,
we require the ability to fetch a list of users based on their external
authentication identifier.

### CODE REVIEW

**Comment by marcparadise** [comment]
I think there's some work needed in the query and possible `chef_user` to support this?

**Comment by chefsalim** [comment]
@marcparadise can you give clarify your comment on the more work?

**Comment by stevendanna** [comment]
The chef_user module already has a handler for external_authentication_ui being set in the StubRec:

```
list(#chef_user{external_authentication_uid = ExtAuthUid}, CallbackFun) when ExtAuthUid =/= undefined ->
    CallbackFun({list_users_by_ext_auth_uid, [ExtAuthUid], [username]});
list(#chef_user{email = undefined} = User, CallbackFun) ->
    CallbackFun({list_query(User), [], [username]});
list(#chef_user{email = EMail}, CallbackFun) ->
    CallbackFun({list_users_by_email, [EMail], [username]}).
```

So this is good to go I believe.  It will ignore the email parameter if that is supplied along with the external_authentication_uid, maybe we want to return an error if both are specified? Doesn't seem like a common use case since we are the only people who use this.

**Comment by marcparadise** [comment]
Disregard - I took a look at the place I thought you'd need to modify (chef_user.erl, pgsql_statements,.config) and saw that we already support it, and it was just a matter of exposing as is done here.

**Comment by stevendanna** [comment]
The failing test is because this endpoint hasn't also been implemented in chef-zero.  Options are to either implement it there or to add a tag to your rspec tests so that you can turn off those tests when run against chef-zero.

**Comment by stevendanna** [comment]
Overall 👍. Before merging y'all will need to make a call on how to handle the chef-zero stuff and whether to return an error when both email and external_authentication_uid is specified.

**Comment by marcparadise** [comment]
:+1: once chef-zero is handled.

File: oc-chef-pedant/spec/api/user_spec.rb [hunk 1]
```diff
@@ -64,6 +64,52 @@
...
+          before :each do
+            @user = platform.create_user(username, user_options)
+          end
+
+          after :each do
+            platform.delete_user(@user)
+          end
+
+          it "returns no users when filtering by non-existing external_authentication_uid", :smoke do
+            get("#{request_url}?external_authentication_uid=somenonexistingemail@somewhere.com", platform.superuser).should look_like({
```
Comment (srenatus) [inline oc-chef-pedant/spec/api/user_spec.rb]: Nitpick, looking at [chef-manage's test stubs](https://github.com/chef/chef-manage/blob/master/src/chef-manage/spec/controllers/sessions_controller_spec.rb#L49-L50), perhaps this should be `%40`... doesn't matter much for this test, though.

**Comment by srenatus** [comment]
LGTM. @stevendanna: +1 re: supporting both filters -- we just don't need that now, so let's fix that when it's needed.
